js/create_progress_bar.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @param     parameters
+ * @param     {HTMLElement} parameters.container Mandatory. The progress bar's unique key.
+ * @param     {string} parameters.key Mandatory. The progress bar's unique key.
+ * @param     {null|function} parameters.progress_callback The function that will be called for each progress response. If the return value is "false", this stops the progress checks.
+ * @param     {null|function} parameters.error_callback The function that will be called for each error, either exceptions or non-200 HTTP responses. Stops the progress checks by default, unless you return a true-ish value from the callback, or unless the error is non-recoverable and implies stopping
+ * @return    {{start: function, stop: function}}
+ */
+function create_progress_bar(parameters)
```

Comment:
That was my first idea, but the problem with that is that it sort of "hides" some other parameters.
For instance, with the refactorings, I could make sure we don't need the "progress_callback". Should I remove the argument then? It's not that easy, because so far it works like that and I am not yet sure about how it's gonna be used later in other progress bars (like batch actions). There are still mandatory parameters, and the jsdoc does all the rest.
If it was typescript, the `parameters` argument would also be self-documented.

I'm still not sure about proper args in this specific case. The function acts like a "library", and for maximum flexibility, an object as param is easier to maintain than a list of arguments that might grow or change.

Reply:
A single config object is the better solution here. JSDoc is also plenty sufficient for modern IDEs to document the object shape and types.

You should declare the object parameter using destructuring to handle default values.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment